### ISSUE TITLE
Is this supposed to work on i3wm?

### ISSUE THREAD

**Issue by justdanyul** [rbreaves/kinto#528]
I just gave this a spin on i3wm, and I didn't have much luck. In short, it didn't seem to work at all. Kinto shows up in the task bar, and seems to be running, after I did the keyboard identification step. But, all my mappings are exactly as they always where. Does anybody actually run this under i3wm? and is it supposed to work?

**Comment by one-cyber-camel** [comment]
I am also trying to get this to work using i3.
When using the install script I get the window to pick the second key beside the spacebar and after that kinto is working.

After reboot it stops working. I have tried to start it using kinto.sh and the kinto-gui.py but no luck.

do you have any tips on what I might try?

**Comment by rbreaves** [comment]
So the Kinto-gui.py fails to load after a reboot? I’d run the ```~/.config/kinto/kinto-gui.py``` & give me the output.

you can also run ```xkeysnail ~/.config/kinto/kinto.py``` if there’s no xkeysnail process running. Probably best to do sudo systemctl stop xkeysnail first. Really there’s plenty of troubleshooting steps in the ReadMe that you’ve not yet gone though. I don’t think I’m mentioning anything new.

**Comment by justdanyul** [comment]
I think in my case, its having problems with the keyboard then. I'm using a UK layout mac keyboard from WASD.  I'll fiddle around with it sometime. But, the short is, if i install it, and start kinto, open Firefox for example, 

I still need CTRL-C to copy and CTRL-P to paste. I got a xmodmap workaround that just flips command and control (i obviously disabled this before trying kinto) , which to be fair, works OK for me. So, I suppose this is more informative for you. On my system, this didn't work out of the box.

**Comment by rbreaves** [comment]
No, that's helpful feedback @justdanyul.. some users and use cases may only want to do a VERY simple key swap of the modifier keys and nothing else. I will seriously consider a future update to allow for 2 additional configurations, base level keyswaps only (no terminal keyswapping), and base level with terminal keyswaps. In both cases no additional muckery of other applications.

Also my readme never mentions i3, although some users appear to be using it with that ok I believe. Also @one-cyber-camel never got back with me but I assume his DISPLAY is something besides 0 and he just needs to update the xkeysnail.service file to use his DISPLAY var, could be your issue as well. I need to be setting that better - generally most DEs use 0 by default unless you have mutliple users going on. Initially I didn't set it at all but that created issues on some distros/DEs.

**Comment by jrbenriquez** [comment]
Same, mine always work on fresh install then on reboot it does not work

Like @rbreaves mentioned I ran
`sudo xkeysnail ~/.config/kinto/kinto.py` once

then I added this to my i3 config
`sudo systemctl start xkeysnail`

**Comment by joshgoebel** [comment]
If you get it working (at all) you may need to go to some trouble to reconfigure parts of it for what keys you intend to use for i3... I was using Cmd as my modifier for Awesome and after Kinto now that I have Cmd back (for Mac keys) my awesome keys were all over the place - so I just create an entirely new modifier and how I user Hyper (via Caps Lock) for all the Window Management stuff - and the Kinto key bindings for everything else.

And since Kinto doesn't know/care about hyper - it's left entirely alone for Awesome WM.

**Comment by webern** [comment]
@joshgoebel 

> And since Kinto doesn't know/care about hyper - it's left entirely alone for Awesome WM.

So kinto "just works" with Awesome as long as your window-manager-modifier key does not conflict with kinto?

**Comment by joshgoebel** [comment]
Well I'm not using Kinto (whole) just the keymappings, and with my own port of the keymapper. So I'd hate to generalize too much.  If you use keyszer - which allows custom modifiers - then it seems to work mostly fine.
